ui/app/store/actions.js

```diff
@@ -17,14 +17,24 @@ import { getEnvironmentType } from '../../../app/scripts/lib/util'
 import * as actionConstants from './actionConstants'
 import {
   getPermittedAccountsForCurrentTab,
+  getSelectedAccount,
   getSelectedAddress,
+  getSwapAmount,
+  getSwapConversionRate,
+  getSwapFromAsset,
+  getSwapFromTokenAssetBalance,
+  getSwapFromTokenContract,
+  getSwapPrimaryCurrency,
+  getSwapQuoteEstimatedGasCost,
 } from '../selectors'
 import { switchedToUnconnectedAccount } from '../ducks/alerts/unconnected-account'
 import { getUnconnectedAccountAlertEnabledness } from '../ducks/metamask/metamask'
-import { dispatch } from 'd3-dispatch'
```

Comment:
If this is deleted, what dispatch are you using?

Reply:
I don't think we need that. The [Redux Thunk middleware](https://github.com/reduxjs/redux-thunk) already provides a `dispatch` to the _action-creators_.